include/utils.h

```diff
@@ -134,6 +134,8 @@ G_GNUC_WARN_UNUSED_RESULT;
 
 /**
  * Get string argument from key and remove key from key_file.
+ *
+ * @return A newly allocated string or NULL.
```

Comment:
Maybe even more clearer with:

``` suggestion
 * @return A newly allocated string or NULL on error.
```

In a follow up we should then also add documentation for the parameters, but I would not see that as the blocker for this initial enhancement ;)